src/plugins/intel_gpu/src/graph/impls/cpu/reorder.cpp

```diff
@@ -56,7 +56,7 @@ struct reorder_impl : public typed_primitive_impl<reorder> {
         auto output_mem_ptr = instance.output_memory_ptr();
 
         cldnn::mem_lock<uint8_t, mem_lock_type::read> input_lock(input_mem_ptr, stream);
-        cldnn::mem_lock<uint8_t, mem_lock_type::read> output_lock(output_mem_ptr, stream);
+        cldnn::mem_lock<uint8_t, mem_lock_type::read_write> output_lock(output_mem_ptr, stream);
```

Comment:
output memory will be updated as the reason solved by this PR, could you descript the detail reason?

Reply:
who allocate cl_mem output for this node? and why cpu impl is choosen?
btw, this line will break if output is usm_device, it's unsafe to directly change to read_write here

Reply:
Sure, River,  I will add some test case to explain this

Reply:
> usm_device

If it is usm_device, it will do copying operation by mem_lock itself, right? But we need logic to write back to usm_device memory if needed.

Reply:
@WeldonWangwang, could you please update access modifiers for the other CPU impls as well?

Reply:
@sshlyapn Sure, will add it.

Reply:
If the memory is used for readonly, why change them to read_write?

Reply:
@riverlijunjie, why are they used for read-only? These buffers are used to store CPU implementations output results, so they should be modifiable